### ISSUE TITLE
Support @RequestPart annotation

### ISSUE THREAD

**Issue by ammmze** [martypitt/swagger-springmvc#588]
TLDR: `@RequestPart` parameters should have `paramType` of `form` and models should be generated for `@RequestPart` annotated parameters.

---

When using the `@RequestPart` annotation, the paramType attribute for that parameter is incorrect (unless the parameter is a `MultipartFile`). When making an api operation that uses a file upload, we can no longer use `@RequestBody`, instead it must be `@RequestPart`, which means the overall request `Content-Type` becomes `mulitpart/form-data`.

While this all works fine if the if the file upload is the only parameter, once you try to add additional parameters (i.e. pieces that will be included in its own boundary), it no longer works (at least not with swagger-ui, because the spec is incorrect).

The other issue is the parameters annotated as `@RequestPart` do not have their models added to the generated spec files.

So given the following example, swagger-ui should create a form that takes a text input and file input (which it does), however the request parameter is using paramType 'body', and when the form is submitted, only the file gets submitted in the request.

```java
@ApiOperation(value = "")
@RequestMapping(value = "/issue", method = RequestMethod.POST)
@ResponseBody
public MyResponse issueRequest(
        @ApiParam(required = true) @RequestPart MyRequest request,
        @ApiParam(required = true) @RequestPart MultipartFile document)
```

Using the custom annotation readers, I am able to work around the paramType issue, but not adding the model to the spec issue.

**Comment by zerkowsm** [comment]
It seems Swagger does NOT display Model | Example value section for @RequestPart in 2.7.0 version. It just displays undefined as data Type next to the text field. It works if you change @RequestPart into @RequestBody.

**Comment by dilipkrish** [comment]
@zerkowsm this is fixed via #1965 in the next release

**Comment by zerkowsm** [comment]
Thanks for the information @dilipkrish . When do you plan to release that?

**Comment by dilipkrish** [comment]
Hopefully in a week. trying to round out some bugs before that.

**Comment by zerkowsm** [comment]
:+1:

**Comment by ndorigatti** [comment]
Hello, i see again this issue in springfox-boot-starter:3.0.0

I have an endpoint with both Pojo and Multipart file.
I can successfully call the endpoint with Postman tool if I set it:
` @PostMapping(value = "/sito", consumes = {MediaType.MULTIPART_FORM_DATA_VALUE})
    public void insertSito(@RequestPart(name = "edifici", required = false) MultipartFile file,
                           @RequestPart(name="sito") CreateSitoDTO creationRequest) {`

But swagger shows ONLY the file part (that is the only optional one btw...).
If i switch the json part to @RequestBody, swagger shows both the elements, but fails to call the endpoint, 
what i get is:
`org.springframework.web.servlet.handler.AbstractHandlerExceptionResolver: Resolved [org.springframework.web.HttpMediaTypeNotSupportedException: Content type 'multipart/form-data;boundary=----WebKitFormBoundaryr6riTHMZjZyrkhQI;charset=UTF-8' not supported]`

I'm currently using spring-boot 2.5.7 and springfox-boot-starter 3.0.0 (NOT snapshot)

**Comment by spasaria** [comment]
Someone solved the problem described by @ndorigatti? i have the same scenario and I don't know how to fix it
